### PR TITLE
feat: hook to auto_complete command

### PR DESCRIPTION
**Pull request by timfjord**
When I started working on this PR my idea was to only add `hook_to_auto_complete_command` settings so the copilot completion appears after clicking `ctrl+space`.  
But then I realised that our current approach of handling the `modified` event breaks the undo functionality(it undos symbol by symbols instead of changes) so this PR also fixes that

### CODE REVIEW

File: sublime-package.json [hunk 1]
```diff
@@ -24,6 +24,11 @@
                       "description": "Auto ask the server for completions. Otherwise, you have to trigger it manually.",
                       "type": "boolean"
                     },
+                    "hook_to_auto_complete_command": {
+                      "default": true,
+                      "description": "Ask the server for completions when the auto_complete command is called.",
```
Comment (jfcherng) [inline sublime-package.json:29]:
```suggestion
                      "markdownDescription": "Ask the server for completions when the `auto_complete` command is called.",
```

File: listeners.py [hunk 2]
```diff
@@ -41,6 +55,15 @@ def test(value: Any) -> Optional[bool]:
             return test(CopilotPlugin.get_account_status().is_authorized)
         return None
 
+    def on_post_text_command(self, command_name: str, args: dict):
```
Comment (jfcherng) [inline listeners.py:58]:
```suggestion
    def on_post_text_command(self, command_name: str, args: Optional[Dict[str, Any]]) -> None:
```

File: LSP-copilot.sublime-settings [hunk 3]
```diff
@@ -12,6 +12,7 @@
 	"initializationOptions": {},
 	"settings": {
 		"auto_ask_completions": true,
+		"hook_to_auto_complete_command": false,
```
Comment (jfcherng) [inline LSP-copilot.sublime-settings:15]: I guess this is defaulted to `true`, as specified in `sublime-package.json`.

File: listeners.py [hunk 4]
```diff
@@ -41,6 +55,15 @@ def test(value: Any) -> Optional[bool]:
...
         return None
 
+    def on_post_text_command(self, command_name: str, args: Optional[Dict[str, Any]]) -> None:
+        if command_name != "auto_complete":
+            return
+
+        _, session = self._get_session()
+
+        if session and get_setting(session, "hook_to_auto_complete_command"):
+            self.view.run_command("copilot_ask_completions")
```
Comment (timfjord) [inline listeners.py:65]: I just realsied, that I can do `plugin.request_get_completions(self.view)` directly here

File: LSP-copilot.sublime-settings (hunk 3 above)
Comment (timfjord) [reply LSP-copilot.sublime-settings:15]: Hm, there is a typo in `sublime-package.json` as my idea was to make it false by default


But I am happy to make it `true`

File: listeners.py (hunk 4 above)
Comment (jfcherng) [reply listeners.py:65]: Seems ok. `request_get_completions` has authorization check.

File: LSP-copilot.sublime-settings (hunk 3 above)
Comment (jfcherng) [reply LSP-copilot.sublime-settings:15]: Since it's configurable, I think the default is not really important. Just to sync two side either way.

Comment (timfjord) [reply LSP-copilot.sublime-settings:15]: Made it `false`

File: listeners.py [hunk 5]
```diff
@@ -8,17 +12,27 @@
...
         if not plugin:
-            return
+            return None, None
 
         session = plugin.weaksession()
         if not session:
-            return
+            return plugin, None
+
+        return plugin, session
```
Comment (jfcherng) [inline listeners.py:24]: Maybe just

```py
return plugin, plugin.weaksession()
```

Comment (timfjord) [reply listeners.py:24]: even shorter, I like it 👍🏻

**Review by jfcherng — APPROVED** [review]

**Review by TheSecEng — APPROVED** [review]
LTGM

**Pull request merged**